### PR TITLE
HDDS-11785. DataNode aborts ContainerStateMachine if it does not know any follower next index

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?
HDDS-11785. DataNode aborts state machine because ContainerStateMachine does not know follower's next index

Please describe your PR in detail:
* When waitOnAllFollowers is enabled (hdds.datanode.wait.on.all.followers), the leader may not have the follower's information, and will throw a RuntimeException that closes the state machine.
* Add a fix to return 0 in that case.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-11785

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java [hunk 1]
```diff
@@ -1006,7 +1006,7 @@ private void removeStateMachineDataIfNeeded(long index) {
         RaftServer.Division division = ratisServer.getServer().getDivision(getGroupId());
         if (division.getInfo().isLeader()) {
           long minIndex = Arrays.stream(division.getInfo()
-              .getFollowerNextIndices()).min().getAsLong();
+              .getFollowerNextIndices()).min().orElse(0);
           LOG.debug("Removing data corresponding to log index {} min index {} "
                   + "from cache", index, minIndex);
           removeCacheDataUpTo(Math.min(minIndex, index));
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java:1012]: Thanks @jojochuang for the patch.

Should it `removeCacheDataUpTo` only if `minIndex` is found?

```java
Arrays.stream(division.getInfo().getFollowerNextIndices())
    .min()
    .ifPresent(minIndex -> removeCacheDataUpTo(Math.min(minIndex, index)));
```

Also, would be nice to rethrow unchecked exceptions directly:

```java
try {
  ...
} catch (RuntimeException e) {
  throw e;
} catch (Exception e) {
  throw new RuntimeException(e);
}
```

Comment (jojochuang) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java:1012]: I think it's essentially the same, but your suggest is more readable. Will update accordingly.

**Review by adoroszlai — COMMENTED** [review]
Thanks @jojochuang for updating the patch, LGTM.

(I think line wrap in the suggestion was more readable, but functionally it's the same, so it's OK.)

**Comment by jojochuang** [comment]
Merged. Thanks Attila

**Pull request merged**